csharp/src/Google.Protobuf/FieldMaskTree.cs

```diff
@@ -333,15 +333,23 @@ public void Merge(IMessage source, IMessage destination, FieldMask.MergeOptions
                         {
                             if (sourceField != null)
                             {
-                                var sourceByteString = ((IMessage)sourceField).ToByteString();
-                                var destinationValue = (IMessage)field.Accessor.GetValue(destination);
-                                if (destinationValue != null)
+                                // Well-known wrapper types are represented as nullable primitive types, so we do not "merge" them.
```

Comment:
```suggestion
                                // Well-known wrapper types are represented as nullable primitive types, so we do not "merge" them.
                                // Instead, any non-null value just overwrites the previous value directly.
```

(Just trying to avoid others having to go through the same mental process I did!)